packages/arb-token-bridge-ui/src/hooks/TransferPanel/useGasEstimates.ts

```diff
@@ -40,44 +45,51 @@ export function useGasEstimates({
   destinationChainId,
   sourceChainErc20Address,
   destinationChainErc20Address,
-  amount
+  amount,
+  balance
 }: {
   walletAddress?: string
   sourceChainId: number
   destinationChainId: number
   sourceChainErc20Address?: string
   destinationChainErc20Address?: string
   amount: BigNumber
+  balance: BigNumber | null
 }): {
   gasEstimates: GasEstimates | DepositGasEstimates | undefined
   error: any
 } {
   const { data: signer } = useSigner()
 
   const { data: gasEstimates, error } = useSWR(
-    typeof signer === 'undefined'
-      ? null
-      : [
-          walletAddress,
+    signer && balance && balance.gt(amount)
```

Comment:
This check skip the gas estimate if user doesn't have enough balance. Fetcher could still throw if amount is close to the balance, and amount + gas is superior to the balance.

Reply:
Do we want to ignore `INSUFFICIENT_FUNDS` error if amount is too close to balance? We would still have logs for actual `INSUFFICIENT_FUNDS` if amount is low enough and it should not throw error, but we would not have false positive anymore